xdsl/transforms/convert_onnx_to_linalg.py

```diff
@@ -115,6 +115,35 @@ def match_and_rewrite(self, constant: onnx.Constant, rewriter: PatternRewriter,
         )
 
 
+@dataclass
+class ReshapeOpLowering(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, reshape: onnx.Reshape, rewriter: PatternRewriter, /):
+        # Dynamic shapes not currently supported
+        source_type = reshape.data.type
+        shape_type = reshape.shape.type
+        print(shape_type.get_shape())
+        if isinstance(source_type, TensorType) and isinstance(shape_type, TensorType):
+            source_shape = source_type.get_shape()
+            shape_shape = shape_type.get_shape()
+            if -1 in source_shape or -1 in shape_shape:
+                raise NotImplementedError()
+
+        # Lowering with `allowzero = 1` attribute not supported"
+        if "allowzero" in reshape.attributes and reshape.attributes["allowzero"] == 1:
```

Comment:
```suggestion
        if reshape.allowzero is not None and reshape.allowzero.data.value.data == 1:
```

Reply:
there pyright issues on `data` and `value` on this

Reply:
their*